File: StudyBook.API/Entities/Subject.cs
using System.ComponentModel.DataAnnotations;

namespace StudyBook.API.Entities
{
    public class Subject
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public List<FlashcardSet> FlashcardSets { get; set; } = new List<FlashcardSet>();
        public List<Test> Tests { get; set; } = new List<Test>();
    }
}

File: StudyBook.Web/src/app/subject/model/subject.ts
import { FlashcardSet } from "../components/flashcard/model/flashcard-set"
import { Test } from "../components/test/model/test"

export interface Subject {
    id: number,
    name: string,
    description: string
    flashcardSets: FlashcardSet[],
    tests: Test[]
}
